### PR TITLE
feat(meson): added unit tests

### PR DESCRIPTION
**Pull request by NaddiNadja**
The unit tests can be run by compiling with the meson option 'tests' set to true and then running `meson test` in the build directory.

### CODE REVIEW

File: tests/meson.build [hunk 1]
```diff
@@ -3,37 +3,82 @@
...
+  ],
+  'compare.c': [
+    ['no args', ['compare', '1GB']],
+    ['slba=0 nlb=7', ['compare', '1GB', '--slba', '0', '--nlb', '7']],
+  ],
+  'copy.c': [
+    ['no args', ['copy', '1GB']],
+    ['slba=0 sdlba=8', ['copy', '1GB', '--slba', '0', '--nlb', '7', '--sdlba', '8']],
+    ['slba=8 sdlba=0', ['copy', '1GB', '--slba', '8', '--nlb', '7', '--sdlba', '0']],  ],
+  'delay_identification.c': [],
```
Comment (karlowich) [inline tests/meson.build:31]: We don't need to skip this.
It works with the arguments: `['open',  '1GB']`

**Review by karlowich — CHANGES_REQUESTED** [review]
This is great!
In addition to the comment I left, I think we should add a `make test`.
You can take inspiration from `make install` in the `Makefile`

**Comment by karlowich** [comment]
Additionally, it would be great to have the same functionality for the executables in `xnvme/tools/`

**Review by karlowich — APPROVED** [review]
Looks good!

**Review by ankit-sam — APPROVED** [review]
Looks good, but just wondering if this should include all the tests (say from xnvme.c we are missing a bunch like, set & get features, dsm, log pages etc).

**Comment by karlowich** [comment]
@ankit-sam We are only running the tests that work with the ramdisk. The idea is when we add more functionality to the ramdisk, we can test more things

**Pull request merged**